tests/net/socket/udp/src/main.c

```diff
@@ -752,6 +753,106 @@ void test_so_txtime(void)
 	zassert_equal(rv, 0, "close failed");
 }
 
+void test_so_rcvtimeo(void)
+{
+	struct sockaddr_in bind_addr4;
+	struct sockaddr_in6 bind_addr6;
+	int sock1, sock2, rv;
+	ssize_t recved = 0;
+	struct sockaddr addr;
+	socklen_t addrlen;
+	uint32_t start_time, time_diff;
+
+	struct timeval optval = {
+		.tv_sec = 2,
+		.tv_usec = 500000,
+	};
+
+	prepare_sock_udp_v4(CONFIG_NET_CONFIG_MY_IPV4_ADDR, 55555,
+			    &sock1, &bind_addr4);
+	prepare_sock_udp_v6(CONFIG_NET_CONFIG_MY_IPV6_ADDR, 55555,
+			    &sock2, &bind_addr6);
+
+	rv = bind(sock1, (struct sockaddr *)&bind_addr4, sizeof(bind_addr4));
+	zassert_equal(rv, 0, "bind failed");
+
+	rv = bind(sock2, (struct sockaddr *)&bind_addr6, sizeof(bind_addr6));
+	zassert_equal(rv, 0, "bind failed");
+
+	rv = setsockopt(sock1, SOL_SOCKET, SO_RCVTIMEO, &optval,
+			sizeof(optval));
+	zassert_equal(rv, 0, "setsockopt failed (%d)", errno);
+
+	optval.tv_usec = 0;
+	rv = setsockopt(sock2, SOL_SOCKET, SO_RCVTIMEO, &optval,
+			sizeof(optval));
+	zassert_equal(rv, 0, "setsockopt failed");
+
+	addrlen = sizeof(addr);
+	clear_buf(rx_buf);
+	start_time = k_uptime_get_32();
+	recved = recvfrom(sock1, rx_buf, sizeof(rx_buf),
+			  0, &addr, &addrlen);
+	time_diff = k_uptime_get_32() - start_time;
+
+	zassert_equal(recved, -1, "Unexpected return code");
+	zassert_equal(errno, EWOULDBLOCK, "Unexpected errno value: %d", errno);
```

Comment:
We never use EWOULDBLOCK in the stack, instead having standardized on EAGAIN. Can you please replace all occurrences to preserve consistency?